lib/services/pdf_renderer/layout.js

```diff
@@ -74,9 +78,14 @@ export function autoLayoutItems(doc, componentSizes, printSettings) {
         // the component fits into the print medium
         // lay it out in rows and columns
 
-        // new page on new component
-        // TODO: "component mingling" print setting, for the paper-conscious
-        currentPage += 1
+        // If we enabled component merging, or it's a different size component, we start a new page
+        if (!printSettings.componentMerging || (lastH !== size.h && lastW !== size.w)) {
```

Comment:
Your comment, "we enabled", I think is supposed to be "disabled".

You conditional says "**both** dimensions don't match", but I think it is supposed to be "**either** dimension doesn't match", right? So `||` instead of `&&`.